src/auth-service/utils/create-user.js

```diff
@@ -1064,6 +1107,80 @@ const join = {
       };
     }
   },
+
+  deleteMobileUserData: async (request) => {
+    try {
+      const { body } = request;
+      let { uid } = body;
+      let { ct } = body;
+      const userRecord = await admin.auth().getUser(uid);
+      
+      //get creation time and compare with ct
+      let creationTime = userRecord.metadata.creationTime;
+      creationTime = creationTime.replace(/\D/g, "");
+      if (creationTime !== ct) {
+        return {
+          success: false,
+          message: "Invalid request",
+          status: httpStatus.BAD_REQUEST,
+          errors: { message: "Invalid request" },
+        };
+      }
+
+      try {
+        await getAuth().deleteUser(uid);
+        const collectionList = [
+          constants.FIREBASE_COLLECTION_KYA,
+          constants.FIREBASE_COLLECTION_ANALYTICS,
+          constants.FIREBASE_COLLECTION_NOTIFICATIONS,
+          constants.FIREBASE_COLLECTION_FAVORITE_PLACES
+        ];
+        let collectionRef = db.collection(`${constants.FIREBASE_COLLECTION_USERS}`);
+        let docRef = collectionRef.doc(uid);
+
+        docRef.delete().then(async () => {
+          for (var collection of collectionList) {
+            await deleteCollection(db, `${collection}/${uid}/${uid}`, 100);
+            collectionRef = db.collection(`${collection}`);
+            docRef = collectionRef.doc(uid);
+            docRef.delete();
+          }
+          console.log('Document successfully deleted!');
+        }).catch((error) => {
+          console.error('Error deleting document:', error);
+          return {
+            success: false,
+            message: "Error deleting Firestore documents",
+            status: httpStatus.INTERNAL_SERVER_ERROR,
+            errors: { message: error.message }
+          };
+        });
+
+        
+
+        return {
+          success: true,
+          message: "User account has been deleted.",
+          status: httpStatus.OK,
+        };
+      } catch (error) {
+        console.error("Error deleting user:", error);
```

Comment:
same here....we could also add the Slack logger. Example: `logger.error(`internal server error -- ${error.message}`);`

Reply:
I've updated these also

Reply:
thanks, we could remove all usages of console functions since we need to globally manage them from the `utils/log` file.  @BenjaminSsempala

Reply:
Thanks @Baalmart I've updated this.